### PR TITLE
SE.Redis DC backend / HybridCache : add net8.0 TFM

### PR DESCRIPTION
**Pull request by mgravell**
Context: Aspire has Garnet support via `AddGarnet()`; Aspire targets LTS, so 8.0; we would like Aspire's Garnet support to work for more than just the raw `IConnectionMultiplexer` usage; specifically, we would like at least some of the inbuilt cache implementations to work. Prior to 9.0, the redis-based `IDistributedCache` does *not* work with Garnet, because it uses Lua, which Garnet does not support. It is not currently planned to backport the necessary Garnet support to 8.0.

In theory, the pre-existing NS targets on these libraries means that 8.0 apps can consume the 9.0 OOB package and work, but we have been advised to avoid the NS target here, as crossing the streams may cause some assembly load problems at runtime. The suggestion is to add an 8.0 TFM to the 9.0 build, which is unusual but should be valid. In particular, this applies to:

1. Microsoft.Extensions.Caching.StackExchangeRedis, which adds Garnet support by switching from Lua to direct commands
2. Microsoft.Extensions.Caching.Hybrid, new feature in 9.0

(note that the "runtime" piece i.e. Microsoft.Extensions.Caching.Abstractions already has an explicit 8.0 TFM in the 9.0 build)

Both of these already have down-level targets, so this is not "new", just a new explicit TFM in the existing matrix.

To avoid leaving 8.0 in hard-to-find places, I have also introduced `$(CurrentLtsTargetFramework)` via `eng/Versions.props`; I have validated that this continues to work correctly when it matches the `$(DefaultNetCoreTargetFramework)`

/cc @mitchdenny

### CODE REVIEW

**Comment by mgravell** [comment]
probably also /cc @wtgodbe in case this is a "known bad thing, verboten"

**Review by halter73 — APPROVED** [review]
Seems reasonable to me, but I'd let @wtgodbe take a look before merging.

File: src/Caching/StackExchangeRedis/src/PublicAPI/net8.0/PublicAPI.Shipped.txt
Comment (wtgodbe) [inline src/Caching/StackExchangeRedis/src/PublicAPI/net8.0/PublicAPI.Shipped.txt:2]: This feels wrong, the new APIs should be the `unshipped.txt` files until we ship 9.0 RTM

File: eng/Versions.props
Comment (wtgodbe) [inline eng/Versions.props:50]: Can you add a bullet point to https://github.com/dotnet/aspnetcore/blob/main/docs/UpdatingMajorVersionAndTFM.md#updating-tfm about updating this?

File: src/Caching/Hybrid/src/Microsoft.Extensions.Caching.Hybrid.csproj
Comment (wtgodbe) [inline src/Caching/Hybrid/src/Microsoft.Extensions.Caching.Hybrid.csproj:12]: Is the plan for this project to _always_ support the current TFM + the most recent LTS TFM, or should this change be undone when we go to `net10.0`?

**Review by wtgodbe — APPROVED** [review]
I'm a little wary of this given that it's something we haven't done before, but given that this is an OOB package & not a library that's in the SharedFx, it's probably not too risky. Approved modulo a few questions

File: src/Caching/StackExchangeRedis/src/PublicAPI/net8.0/PublicAPI.Shipped.txt
Comment (mgravell) [reply src/Caching/StackExchangeRedis/src/PublicAPI/net8.0/PublicAPI.Shipped.txt:2]: fixed; agreed this is "unshipped"

File: src/Caching/Hybrid/src/Microsoft.Extensions.Caching.Hybrid.csproj
Comment (mgravell) [reply src/Caching/Hybrid/src/Microsoft.Extensions.Caching.Hybrid.csproj:12]: We can leave this alone - for LTS builds the `DefaultNetCoreTargetFramework` and `CurrentLtsTargetFramework` will match, meaning it gets de-dup'd and we just get the LTS target (plus the spares)

**Pull request merged**
